en/flight_controller/beaglebone_blue.md

```diff
@@ -17,118 +17,114 @@ Information about flashing OS images can be found on [this page](https://github.
+   For *rsync* over SSH with key authentication, follow steps here:
+   1. sudo su
+   1. ssh-keygen -t rsa -C root@beaglebone
+   1. ENTER //no passphrase
+   1. ENTER
+   1. ENTER
+   1. ssh-copy-id root@beaglebone
+   1. When promted if you trust - yes
+   1. password root
+   1. On the development computer, define the BeagleBone Blue board as `beaglebone` in **/etc/hosts**
+1. Cross Compile Setup
+   1. Toolchain download
+      1. First install the toolchain into */opt/bbblue_toolchain/gcc-arm-linux-gnueabihf*.
+         Here is an example of using soft link to select which version of the toolchain you want to use:
+         ```
+         mkdir -p /opt/bbblue_toolchain/gcc-arm-linux-gnueabihf
+         chmod -R 777 /opt/bbblue_toolchain
+         ```
+         ARM Cross Compiler for *BeagleBone Blue* can be found at [Linaro Toolchain Binaries site](http://www.linaro.org/downloads/). 
+  
+   
+         > **Tip** GCC in the toolchain should be compatible with kernel in *BeagleBone Blue*.
+            General rule of thumb is to choose a toolchain where version of GCC is not higher than version of GCC which comes with the OS image on *BeagleBone Blue*. 
+   
+         Download and unpack [gcc-linaro-7.5.0-2019.12-x86_64_arm-linux-gnueabihf](https://releases.linaro.org/components/toolchain/binaries/latest-7/arm-linux-gnueabihf/gcc-linaro-7.5.0-2019.12-x86_64_arm-linux-gnueabihf.tar.xz) to the bbblue_toolchain folder
+         ``` 
+         sh
+         ...@ubuntu:/opt/bbblue_toolchain$ ls -l
+         lrwxrwxrwx 1 root root   51 Mar 22 16:10 gcc-arm-linux-gnueabihf -> gcc-linaro-6.3.1-2017.05-x86_64_arm-linux-gnueabihf
+         drwxr-xr-x 8 root root 4096 May 17  2017 gcc-linaro-6.3.1-2017.05-x86_64_arm-linux-gnueabihf
+         drwxr-xr-x 8 root root 4096 Nov 19 03:51 gcc-linaro-6.4.1-2017.11-x86_64_arm-linux-gnueabihf
+         ```
+         The following is an example setup on development host. 
+      1. Add it to the PATH in ~/.profile as shown below
+         ```sh
+         export PATH=$PATH:/opt/bbblue_toolchain/gcc-arm-linux-gnueabihf/gcc-linaro-6.3.1-2017.05-x86_64_arm-linux-gnueabihf/bin
+         export CrossCompiler=/opt/bbblue_toolchain/gcc-arm-linux-gnueabihf/gcc-linaro-6.3.1-2017.05-x86_64_arm-linux-gnueabihf/bin/arm-linux-    gnueabihf-
+         ```
+         ```
+         sudo usermod -a -G dialout $USER
+         ```
+         > **Note** Logout and Login to make the change working
+   
+         Download script for Ubuntu and run it
+         ```
+         curl https://raw.githubusercontent.com/PX4/Devguide/v1.9.0/build_scripts/ubuntu_sim_ros_melodic.sh --output ubuntu_sim_ros_melodic.sh
+         chmod 777 ubuntu_sim_ros_melodic.sh
+         apt-get install git
+         mkdir ~/src
+         cd ~/src
+         git clone https://github.com/PX4/Firmware.git
+         cd Firmware
+         apt install python3-pip
+         pip3 install --user empy
```

Comment:
These should be done from the install script.

Reply:
on Ubuntu 18.04 LTS it was not so, so i had to do it manual.

Reply:
I think @bkueng is suggesting that you can add that stuff into the build script yourself. That would be a separate PR.

Reply:
I think @bkueng is suggesting that you can add that stuff into the build script yourself. That would be a separate PR.